Trademark guidelines update (#4030)
jenkins-infra/jenkins.io: content/projects/gsoc/mentors.adoc

** FINE, but ask GSoC Org Admins to contact the company. Maybe they agree to open-source it (and to assign a mentor). If no, it's their problem.
* "I want to implement a feature based on a patented algorithm/technology. It's open-source, so we are free to do whatever"
** NOT FINE, Jenkins project recognizes laws.

> We are under umbrella of the Linux Foundation subject for US and international legislation.

Reviewer: Minor phrasing change?
```suggestion
We are under the umbrella of the Linux Foundation, subject to US and international law.
```
